modules/improvedigitalBidAdapter.js

```diff
@@ -286,529 +223,374 @@ export const spec = {
-    }
-    if (placementObject.placementKey) {
-      impressionObject.pkey = placementObject.placementKey;
-    }
-    if (placementObject.transactionId) {
-      impressionObject.tid = placementObject.transactionId;
-    }
-    if (!isEmpty(placementObject.video)) {
-      const video = Object.assign({}, placementObject.video);
-      // skip must be 0 or 1
-      if (video.skip !== 1) {
-        delete video.skipmin;
-        delete video.skipafter;
-        if (video.skip !== 0) {
-          logWarn(`video.skip: invalid value '${video.skip}'. Expected 0 or 1`);
-          delete video.skip;
-        }
-      }
-      if (!isEmpty(video)) {
-        impressionObject.video = video;
+      const site = {};
+      const url = config.getConfig('pageUrl') || deepAccess(bidderRequest, 'refererInfo.referer');
+      if (url) {
+        site.page = url;
+        site.domain = parseUrl(url).hostname;
       }
+      const configSiteSettings = config.getConfig('site') || {};
+      const fpdSiteSettings = config.getConfig('ortb2.site') || {};
+      mergeDeep(site, configSiteSettings, fpdSiteSettings);
+      request.site = site;
     }
-    if (placementObject.keyValues) {
-      for (let key in placementObject.keyValues) {
-        for (let valueCounter = 0; valueCounter < placementObject.keyValues[key].length; valueCounter++) {
-          impressionObject.kvw = impressionObject.kvw || {};
-          impressionObject.kvw[key] = impressionObject.kvw[key] || [];
-          impressionObject.kvw[key].push(placementObject.keyValues[key][valueCounter]);
-        }
+  },
+};
+
+const ID_RESPONSE = {
+  buildAd(bid, bidRequest, bidResponse) {
+    if (bidRequest.mediaTypes && Object.keys(bidRequest.mediaTypes).length === 1) {
+      if (deepAccess(bidRequest, 'mediaTypes.video')) {
+        this.buildVideoAd(bid, bidRequest, bidResponse);
+      } else if (deepAccess(bidRequest, 'mediaTypes.banner')) {
+        this.buildBannerAd(bid, bidRequest, bidResponse);
+      } else if (deepAccess(bidRequest, 'mediaTypes.native')) {
+        this.buildNativeAd(bid, bidRequest, bidResponse)
+      }
+    } else {
+      if (bidResponse.adm.search(/^<vast/i) === 0) {
+        this.buildVideoAd(bid, bidRequest, bidResponse);
+      } else if (bidResponse.adm.indexOf('{') === 0) {
+        this.buildNativeAd(bid, bidRequest, bidResponse);
+      } else {
+        this.buildBannerAd(bid, bidRequest, bidResponse);
       }
     }
+  },
 
-    impressionObject.banner = {};
-    if (placementObject.size && placementObject.size.w && placementObject.size.h) {
-      impressionObject.banner.w = placementObject.size.w;
-      impressionObject.banner.h = placementObject.size.h;
+  buildVideoAd(bid, bidRequest, bidResponse) {
+    bid.mediaType = VIDEO;
+    bid.vastXml = bidResponse.adm;
+    if (ID_REQUEST.isOutstreamVideo(bidRequest)) {
+      bid.adResponse = { content: bid.vastXml };
+      bid.renderer = ID_OUTSTREAM.createRenderer(bidRequest);
     }
+  },
```

Comment:
Hmm I think this update inadvertently drops width/height definition for video bids which is a required response field as per https://docs.prebid.org/dev-docs/bidder-adaptor.html#interpreting-the-response. Can you validate that this is still being set?

Reply:
There has never been a proper size for video bids coming from our ad server. Previously it was defaulted to 1x1 but since bidfactory defaults already to 0x0, the default size was removed and is 0x0 now in the final bid.
 https://github.com/prebid/Prebid.js/blob/master/src/bidfactory.js#L22

Screenshot 2022-03-30 at 17.49.57